presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -533,12 +533,21 @@ public void testSTIntersects()
     public void testSTOverlaps()
     {
         assertFunction("ST_Overlaps(ST_GeometryFromText('POINT (50 100)'), ST_GeometryFromText('POINT (150 150)'))", BOOLEAN, false);
+        assertFunction("ST_Overlaps(ST_GeometryFromText('POINT (50 100)'), ST_GeometryFromText('POINT (50 100)'))", BOOLEAN, false);
         assertFunction("ST_Overlaps(ST_GeometryFromText('MULTIPOINT (50 100, 50 200)'), ST_GeometryFromText('POINT (50 100)'))", BOOLEAN, false);
         assertFunction("ST_Overlaps(ST_GeometryFromText('LINESTRING (0 0, 0 1)'), ST_GeometryFromText('LINESTRING (1 1, 1 0)'))", BOOLEAN, false);
         assertFunction("ST_Overlaps(ST_GeometryFromText('MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))'), ST_GeometryFromText('MULTILINESTRING ((3 4, 6 4), (5 0, 5 4))'))", BOOLEAN, true);
         assertFunction("ST_Overlaps(ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('POLYGON ((3 3, 3 5, 5 5, 5 3))'))", BOOLEAN, true);
+        assertFunction("ST_Overlaps(ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'))", BOOLEAN, false);
+        assertFunction("ST_Overlaps(ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'), ST_GeometryFromText('LINESTRING (1 1, 4 4)'))", BOOLEAN, false);
         assertFunction("ST_Overlaps(ST_GeometryFromText('POLYGON ((1 1, 1 3, 3 3, 3 1))'), ST_GeometryFromText('POLYGON ((4 4, 4 5, 5 5, 5 4))'))", BOOLEAN, false);
         assertFunction("ST_Overlaps(ST_GeometryFromText('MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((0 0, 0 2, 2 2, 2 0)))'), ST_GeometryFromText('POLYGON ((0 1, 3 1, 3 3, 0 3))'))", BOOLEAN, true);
+
+        // TODO: ESRI library bug
+        // ST_Overlaps: Returns TRUE if the Geometries share space, are of the same dimension, but are not completely contained by each other.
```

Comment:
This is the right behavior. The SQL/MM Part 3 spec says that a pre-requisite for ST_Overlaps(a, b) == true is that both geometries are of the same dimension. Here, linestring is one-dimentional while polygon is two-dimentional.

12) ST_Overlaps: tests if an ST_Geometry value spatially overlaps another ST_Geometry value.
Given two geometries a and b, ST_Overlaps is defined as: Case:
a) If (a.ST_Dimension() = 0 (zero) and b.ST_Dimension() = 0 (zero)) or (a.ST_Dimension() = 1 (one) and b.ST_Dimension() = 1 (one)) or (a.ST_Dimension() = 2 and b.ST_Dimension() = 2), then:
a.ST_Overlaps(b) 
(Interior(a).ST_Dimension() = Interior(b).ST_Dimension() =
(Interior(a)  Interior(b)).ST_Dimension())  (a  b  a)  (a  b  b) b) Otherwise, the null value
Expressed in terms of the DE-9IM: Case:
a) If (a.ST_Dimension() = 0 (zero) and b.ST_Dimension() = 0 (zero)) or (a.ST_Dimension() = 2 and b.ST_Dimension() = 2), then:
a.ST_Overlaps(b)  (Interior(a)  Interior(b)  ) 
(Interior(a)  Exterior(b)  ) 
(Exterior(a)  Interior(b)  )  a.ST_Relate(b, 'T*T***T**')
b) If a.ST_Dimension() = 1 (one) and b.ST_Dimension() = 1 (one), then:
© ISO/IEC 2016 – All rights reserved 23
Licensed to Maria Basmanova. ANSI order X_486440. Downloaded 8/31/2017 3:16 PM. Single user license only. Copying and networking prohibited.
ISO/IEC 13249-3:2016(E)
a.ST_Overlaps(b) 
((Interior(a)  Interior(b)).ST_Dimension() = 1 (one)) 
(Interior(a)  Exterior(b)  )  (Exterior(a)  Interior(b)  )  a.ST_Relate(b, '1*T***T**')
c) Otherwise, the null value
When determining if two geometries intersect, z and m coordinate values are ignored in the calculations.